### PR TITLE
Small change in DG : Storage component

### PR DESCRIPTION
**Pull request by yuifuku1118**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -165,12 +165,16 @@ The `Model` component,
...
 The `Storage` component,
-* can save address book data, task book data, sales order book data
+* Saves address book data, task book data, sales order book data
   and user preference data in json format, and read them back into corresponding objects.
-* inherits from all of `AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`,
+* The main storage class inherits from all of `AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`,
   and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
-* depends on some classes in the `Model` component (because the `Storage` component's job is to save/retrieve objects that belong to the `Model`)
-
+* Each Book Storage component (`AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`) has a `JsonSerializable` class which is in charge of making converting the model's data into correct json file and retrieving the data from
```
Comment (GnohChengYi) [inline docs/DeveloperGuide.md:172]: Is the word `making` extra?

```suggestion
* Each Book Storage component (`AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`) has a `JsonSerializable` class which is in charge of converting the model's data into correct json file and retrieving the data from 
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -165,12 +165,16 @@ The `Model` component,
...
+* The main storage class inherits from all of `AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`,
   and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
-* depends on some classes in the `Model` component (because the `Storage` component's job is to save/retrieve objects that belong to the `Model`)
-
+* Each Book Storage component (`AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`) has a `JsonSerializable` class which is in charge of making converting the model's data into correct json file and retrieving the data from 
+  the json file to convert it to a model data.
+* Each `JsonSerializable` class implements its own `JsonAdapted` class which specifies methods to convert model Object
+  (i.e `Person`, `Task`, `Order`) into json object and vise versa. 
+* The `JsonSerializable` class and `JsonAdapted` class also checks the correctness of the json files format, and in the 
+  case when any of the format is wrong, it will then throw a `IllegalValueException` and  `IllegalValueException`
```
Comment (GnohChengYi) [inline docs/DeveloperGuide.md:177]: Why is the `IllegalValueException` repeated?
Also, consider changing the article **a** to **an**.

**Review by GnohChengYi — CHANGES_REQUESTED** [review]
There might be some mistakes.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -165,12 +165,16 @@ The `Model` component,
...
+* The main storage class inherits from all of `AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`,
   and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
-* depends on some classes in the `Model` component (because the `Storage` component's job is to save/retrieve objects that belong to the `Model`)
-
+* Each Book Storage component (`AddressBookStorage`, `TaskBookStorage`, `OrderBookStorage`) has a `JsonSerializable` class which is in charge of converting the model's data into correct json file and retrieving the data from 
+  the json file to convert it to a model data.
+* Each `JsonSerializable` class implements its own `JsonAdapted` class which specifies methods to convert model Object
+  (i.e `Person`, `Task`, `Order`) into json object and vise versa. 
+* The `JsonSerializable` class and `JsonAdapted` class also checks the correctness of the json files format, and in the 
+  case when any of the format is wrong, it will then throw a `DataConversionException` and  `IllegalValueException`
```
Comment (GnohChengYi) [inline docs/DeveloperGuide.md:177]: Observe that this entire chunk is under line 167

> **The `Storage` component,** 

This is why the original bullet points start with small letter verbs like **can save**, **inherits** and **depends**.

Would it be better to change line 167 to "How the `Storage` component works:" (refer to lines 116-120 and lines 133-135)?

**Review by GnohChengYi — CHANGES_REQUESTED** [review]
Thanks for updating. I just noticed a possible improvement.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (yuifuku1118) [reply docs/DeveloperGuide.md]: Thanks for the further suggestion, I have changed it!

**Review by GnohChengYi — APPROVED** [review]

**Pull request merged**